modules/summarize.py

```diff
@@ -155,15 +155,16 @@ def summarize_file(file_path, issue_areas, model="gpt-4o", chunk_size=100000, ov
     Returns:
         str: The final summary of the text.
     """
-    with open(file_path, "r") as file:
+    with open(file_path, "r", encoding="utf-8") as file:
         text = file.read()
 
     summary = summarize_text_all_issues(text, issue_areas, model, chunk_size, overlap)
 
     if save_summary:
-        file_name = file_path.split("/")[-1]
-        summary_file_name = f"../data/summaries/{file_name.split('.')[0]}_summary.txt"
-        with open(summary_file_name, "w") as file:
+        input_filename, _ = os.path.splitext(os.path.basename(file_path))
+        summary_file_name = os.path.join(output_dir, f"{input_filename}_summary.txt")
```

Comment:
I would avoid hardcoding any subfolder paths here. If a subfolder is necessary, it could be included in the "output_dir" when passed in this function.